modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/DefinitionDeliveryMaxSubscriptionCyclesExceptionMapper.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.commerce.admin.catalog.internal.jaxrs.exception.mapper;
+
+import com.liferay.commerce.product.exception.CPDefinitionDeliveryMaxSubscriptionCyclesException;
+import com.liferay.portal.vulcan.jaxrs.exception.mapper.BaseExceptionMapper;
+import com.liferay.portal.vulcan.jaxrs.exception.mapper.Problem;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Joao Cordeiro
+ */
+@Component(
+	property = {
+		"osgi.jaxrs.application.select=(osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Catalog)",
+		"osgi.jaxrs.extension=true",
+		"osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Catalog.DefinitionDeliveryMaxSubscriptionCyclesExceptionMapper"
+	},
+	service = ExceptionMapper.class
+)
+public class DefinitionDeliveryMaxSubscriptionCyclesExceptionMapper
+	extends BaseExceptionMapper
+		<CPDefinitionDeliveryMaxSubscriptionCyclesException> {
+
+	@Override
+	protected Problem getProblem(
+		CPDefinitionDeliveryMaxSubscriptionCyclesException
+			cpDefinitionDeliveryMaxSubscriptionCyclesException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST,
```

Comment:
@alessio-rendina is this right according to the new pattern?

Reply:
Yes, for now it's good. We need to refactor all of them in the future.